pkg/sentry/platform/ring0/pagetables/pagetables.go

```diff
@@ -134,6 +163,15 @@ func (v *unmapVisitor) visit(start uintptr, pte *PTE, align uintptr) {
 //
 //go:nosplit
 func (p *PageTables) Unmap(addr usermem.Addr, length uintptr) bool {
+	if p.upperHalfPageTables != nil {
```

Comment:
Same comments as above:
- Nit: full sentence
- Let's add an overflow check
- halfSize could be e.g. p.upperStart